resource_activity/models/resource_activity.py

```diff
@@ -678,11 +677,18 @@ def prepare_lines(self, activity):
 
     def _create_sale_order(self, activity, partner_id):
         SaleOrder = self.env['sale.order']
+        sale_note = SaleOrder._default_note()
+        for line in activity.location_id.line_ids:
+            if (line.note_id
+                    and line.location_id == activity.location_id
+                    and line.activity_type_id == activity.activity_type):
+                sale_note = line.note_id.content

```

Comment:
could you have different activity type on the different line ? 

- if not just add a break after the first matching.
- if yes it the content of the last line that will be taken ?

Reply:
I'll add a `break` in the loop if the `filtered` option is time consuming